File: correlation_coefficient.py
from math import *
def sum_array(x):
    sum = 0
    for i in x:
        sum += i
    return sum

def sum_x_y(x,y):
    sum = 0
    multi = [0]*len(x)
    for i in range(len(x)):
        multi[i] = x[i] * y[i]
    
    for j in range(len(x)):
        sum += multi[j]
    return sum

def sum_array2(x):
    sum = 0
    for i in range(len(x)):
        sum += pow(x[i],2);
    return sum

def cc(x,y,n):
    return (n*(sum_x_y(x,y))-(sum_array(x)*sum_array(y))) / ((sqrt(n*sum_array2(x)-sum_array(x)**2)*(sqrt(n*sum_array2(y)-sum_array(y)**2))))

x = [1,2,3,4,5,6,7,8]
y = [8,7,6,5,4,3,2,1]
n = 8
print(sum_array(x))
print(sum_array(y))
print(sum_x_y(x,y))
print(sum_array2(x))
print(sum_array2(y))
print(cc(x,y,n))

File: correlation_coefficient.c
#include <stdio.h>
#include <math.h>
float sum_array(int x[],int n)
{
    float sum = 0;
    for(int i = 0 ; i < n ; i++)
    {
        sum += x[i];
    }
    return sum;
}
float sum_x_y(int x[],int y[],int n)
{
    float sum = 0;
    float multi[n];
    for(int i = 0 ; i < n ; i++)
    {
        multi[i] = x[i] * y[i];
    }
    for(int j = 0 ; j < n ; j++)
    {
        sum += multi[j];
    }
    return sum;
}
float sum_array2(int x[],int n)
{
    float sum = 0;
    for(int i = 0 ; i < n ; i++)
    {
        sum += pow(x[i],2);
    }
    return sum;
}
float cc(int x[],int y[], int n)
{
    return (n*(sum_x_y(x,y,n))-(sum_array(x,n)*sum_array(y,n))) / ((sqrt(n*sum_array2(x,n)-pow(sum_array(x,n),2))*(sqrt(n*sum_array2(y,n)-pow(sum_array(y,n),2)))));
}
void main()
{
    int x[] = {1,2,3,4,5,6,7,8};
    int y[] = {8,7,6,5,4,3,2,1};
    int n = 8;
    printf("%f\n",sum_array(x,n));
    printf("%f\n",sum_array(y,n));
    printf("%f\n",sum_x_y(x,y,n));
    printf("%f\n",sum_array2(x,n));
    printf("%f\n",sum_array2(y,n));
    printf("%f\n",cc(x,y,n));
}